### PR TITLE
Rewrite services tests to work with postgres

### PR DESCRIPTION
**Pull request by helfi92**
Resolves https://github.com/taskcluster/taskcluster/issues/2558.

Services to cover:
* [x] secrets
* [x] purge-cache
* [x] auth
* [x] github
* [x] hooks
* [x] notify
* [x] index
* [x] queue
* [x] web-server
* [x] worker-manager

TODO:
* [x] Fix tests in taskcluster-queue. (1 failing - `listArtifacts, listLatestArtifacts (continuationToken)`)
* [x] Fix tests in taskcluster-worker-manager
* [x] Fix tests in taskcluster-auth
* [x] File a bug re:tc-worker-manager test (fake function isn't there)

### CODE REVIEW

**Comment by djmitche** [comment]
@helfi92 I added b734dee06aded581c49cf24da0a55841f091a6ce here, since getting the tests to pass requires basing it on this PR.  I'm going to merge a stripped-down version of this patch (omitting test changes) to the tc-lib-postgres branch.

**Comment by helfi92** [comment]
Will flag @taskcluster/security-folks before merging but for now, I'd like to get an initial review from @djmitche.

File: services/queue/src/dependencytracker.js [hunk 1]
```diff
@@ -228,15 +228,23 @@ class DependencyTracker {
...
     await this.TaskDependency.query(condition, {
       limit: 250,
       handler: async (dep) => {
+        if (resolution !== 'completed') {
+          // If the resolution wasn't 'completed', we can only remove
+          // TaskRequirement entries if the 'require' relation is 'resolved'.
+          if (dep.require !== 'resolved') {
+            return;
+          }
+        }
```
Comment (djmitche) [inline services/queue/src/dependencytracker.js]: Nicely done :)

**Comment by helfi92** [comment]
Actually, this is not merging to master just yet, so security review probably not needed atm.

**Pull request merged**